src/net_processing.cpp

```diff
@@ -819,6 +831,19 @@ static void UpdatePreferredDownload(const CNode& node, CNodeState* state) EXCLUS
     nPreferredDownload += state->fPreferredDownload;
 }
 
+std::chrono::microseconds PeerManagerImpl::NextInvToInbounds(std::chrono::microseconds now,
+                                                             std::chrono::seconds average_interval)
+{
+    AssertLockHeld(cs_main);
+    if (m_next_inv_to_inbounds < now) {
+        // If this function were called from multiple threads simultaneously
+        // it would possible that both update the next send variable, and return a different result to their caller.
+        // This is not possible in practice as only the net processing thread invokes this function.
```

Comment:
Oops. I should have removed this comment since it's obvious from the function signature and the assert above that cs_main is always held when reading/writing this data.